models/remoteworkitem.go

```diff
@@ -0,0 +1,77 @@
+package models
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}
+}
+
+// WorkItemMap : holds the mapping between Remote key and local key in a WI
+type WorkItemMap map[string]interface{}
+
+// RemoteWorkItemSanitizer makes sure that the
+// Response from Provider API is sanitized into a
+// good looking flat 1-level dictionary
+type RemoteWorkItemMapper interface {
+	Flatten()
+	GetWorkItemKeyMap()
+}
+
+// GithubRemoteWorkItem is a derivative of RemoteWorkItem
+type GithubRemoteWorkItem struct {
```

Comment:
Not quite sure why you have GithubRemoteWorkItem with a GetWorkItemKeyMap as oppose to:

```go
var remoteWorkItemMaps = map[string]WorkItemMap{"github":  WorkItemMap{....}}

workItem := MapRemoteWorkItem(remoteWorkItemMaps["github"], workitemtype)
 ```